File: ums.client/src/shared/models/Teacher.ts
export class Teacher {
  id: number;
  creatorUserId: string;
  name: string | null = null;
  creationDate?: Date;
  modificationUserId?: string;
  modificationDate?: Date;
  isDeleted: boolean;
  deletionUserId?: string;
  deletionDate?: Date;

  constructor(
    id: number,
    creatorUserId: string,
    name: string | null = null,
    isDeleted: boolean,
    creationDate?: Date,
    modificationUserId?: string,
    modificationDate?: Date,
    deletionUserId?: string,
    deletionDate?: Date
  ) {
    this.id = id;
    this.creatorUserId = creatorUserId;
    this.name = name;
    this.creationDate = creationDate;
    this.modificationUserId = modificationUserId;
    this.modificationDate = modificationDate;
    this.isDeleted = isDeleted;
    this.deletionUserId = deletionUserId;
    this.deletionDate = deletionDate;
  }
}

File: UMS.Core/Models/Teacher.cs
using System;
using System.Collections.Generic;

namespace UMS.Core.Models;

public partial class Teacher
{
    public int Id { get; set; }

    public string CreatorUserId { get; set; } = null!;

    public string Name { get; set; } = null!;

    public DateTime? CreationDate { get; set; }

    public string? ModificationUserId { get; set; }

    public DateTime? ModificationDate { get; set; }

    public bool IsDeleted { get; set; }

    public string? DeletionUserId { get; set; }

    public DateTime? DeletionDate { get; set; }

    public virtual ICollection<Subject> Subjects { get; set; } = new List<Subject>();
}
